### ISSUE TITLE
Takeover of group formation to Moodle groups is slow in large university environments

### ISSUE THREAD

**Issue by jkonert** [moodlepeers/moodle-mod_groupformation#49]
Event though the takeover of group preview into real Moodle groups creates Moodle groups quite fast, the page itself takes long to load after that click.
TODO: Check which code-lines may do inefficient DB-requests here after the Moodle groups are created.

**Comment by Nullmann** [comment]
the page itself takes long to load after that click.
<cite>jkonert</cite>

Should be solved by #17. The real problem is in fact adding the users to the groups.

This is done by the core function in 
```
E:\GitHub\lokales Moodle\server\moodle\group\lib.php:groups_add_member
```
which uses (multiple?) db calls per user and is thus slow.

Because core functions are there for a reason (e.g. checking things you would never think of) and it is not a fault to use them, we would suggest that running it as a (cron) job in the background is the best solution. 

The user has to be notified, preferably with an estimated guess on how long it will take. Something along the lines:
"Moodle-Groups are being created. It will probably take {{time}} minutes"
where time could be something like 
```
time = round_up_to_minute(amount_of_stundends * time_pe_rstudent + amount_of_groups * time_per_group + puffer_time)
```

**Comment by rroepke** [comment]
For a first improvement - group adoption to moodle groups is now implemented as a cron job instance. Please consider if you pull the newest master version updates are required due to database changes. Please test on dev systems and report if it works better :-)
